src/pages/db/_components/Subnav.astro

```diff
@@ -67,18 +67,23 @@ import Badge from "./Badge.astro";
 	const targets = Array.from(nav.querySelectorAll("a[href^='#']"))
 		.map((a) => a.getAttribute("href")?.slice(1))
 		.filter(Boolean);
+	const selector = `:is(${targets.map((id) => `[id="${id}"]`).join(",")})`;
+	const ids: Record<string, number> = {};
 	const io = new IntersectionObserver(
 		([entry]) => {
-			if (entry.intersectionRatio > 0.5) {
-				const element = entry.target.querySelector(
-					`:is(${targets.map((id) => `[id="${id}"]`).join(",")})`,
-				);
-				if (element) {
-					const link = nav.querySelector(`a[href="#${element.id}"]`);
-					resetSubnav();
-					link?.setAttribute("aria-current", "true");
-				}
+			console.log(
+				entry.target.querySelector(selector),
+				entry.intersectionRatio,
+			);
+			const element = entry.target.querySelector(selector);
+			if (!element) return;
+			const prevRatio = ids[element.id];
```

Comment:
It looks like this could be undefined. JS being JS it probably still works, but I'd prefer a more explicit comparison

```suggestion
			const prevRatio = ids[element.id] ?? 0;
```